src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/Frame.FeatureCollection.cs

```diff
@@ -202,6 +203,28 @@ string IHttpRequestIdentifierFeature.TraceIdentifier
             set => TraceIdentifier = value;
         }
 
+        long? IHttpMaxRequestBodySizeFeature.MaxRequestBodySize
+        {
+            get => MaxRequestBodySize;
+            set
+            {
+                if (HasStartedConsumingRequestBody)
```

Comment:
Right now, the app has to keep track of whether or not it's read from the request body itself. I know this might break down a little if you are writing some reusable component where you want to change the upload limit, but only if nothing has been uploaded yet. That seems pretty niche though, and nothing uses this feature today ofc.

Do you want to add HasStartedConsumingRequestBody to some request feature? Do you want to do the same for WasUpgraded?